profiler/src/trace_reader.rs

```diff
@@ -25,6 +28,48 @@ pub enum SampleType {
 pub struct Sample {
     pub location: Location,
     pub sample_type: SampleType,
+    pub flat_resources: ExecutionResources,
+}
+
+impl Sample {
+    pub fn extract_measurements(
+        &self,
+        measurement_types: &[ValueType],
+        context: &ProfilerContext,
+    ) -> Vec<i64> {
+        let mut measurements_map: HashMap<&str, i64> = vec![
```

Comment:
This is prob not the best way to do it but couldn't think of anything better. For sure it's better than bunch of brute force ifs or coupling parts of code that shouldn't be coupled together (keep in mind measurement values order is dependent on measurement types order)